### PR TITLE
Use uint32_t as cross-platform thread and process id type

### PR DESCRIPTION
**Pull request by pierricgimmig**
Use uint32_t as the common type to represent platform-agnostic thread
and process ids. Define kInvalidThreadId and kInvalidProcessId as
platform-agnostic invalid ids. The native thread and process utils are still
available, but now have the "Native" suffix. Also, add conversion utils
to and from native id types.

See go/stadia-orbit-pids-tids for more info.

Tests:  loading old/new captures, navigating capture window, manual and 
dynamic instrumentation, sampling, thread names, selecting samples.

I suggest reviewing individual commits. It looks massive but the changes are
fairly small and are mainly in the first two commits. Thanks for your help!

### CODE REVIEW

File: src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp [hunk 1]
```diff
@@ -328,7 +328,10 @@ class LinuxTracingIntegrationTestFixture {
  public:
   LinuxTracingIntegrationTestFixture() : puppet_{&LinuxTracingIntegrationTestPuppetMain} {}
 
-  [[nodiscard]] pid_t GetPuppetPid() const { return puppet_.GetChildPid(); }
+  [[nodiscard]] pid_t GetPuppetPidNative() const { return puppet_.GetChildPidNative(); }
+  [[nodiscard]] uint32_t GetPuppetPid() const {
+    return orbit_base::GetThreadIdFromNative(GetPuppetPidNative());
```
Comment (beckerhe) [inline src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp:333]: Shouldn't this call `GetProcessIdFromNative`?

File: src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp [hunk 2]
```diff
@@ -337,7 +340,7 @@ class LinuxTracingIntegrationTestFixture {
   [[nodiscard]] orbit_grpc_protos::CaptureOptions BuildDefaultCaptureOptions() {
     orbit_grpc_protos::CaptureOptions capture_options;
     capture_options.set_trace_context_switches(true);
-    capture_options.set_pid(puppet_.GetChildPid());
+    capture_options.set_pid(puppet_.GetChildPidNative());
```
Comment (beckerhe) [inline src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp:343]: Shouldn't capture options contains the non-native pid?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

But someone else should have a look as well as I'm not super familiar with much of the code.

File: src/ApiLoader/EnableInTracee.cpp [hunk 3]
```diff
@@ -30,7 +30,7 @@ using orbit_user_space_instrumentation::ExecuteInProcess;
 
 namespace {
 
-ErrorMessageOr<absl::flat_hash_map<std::string, ModuleInfo>> GetModulesByPathForPid(int32_t pid) {
+ErrorMessageOr<absl::flat_hash_map<std::string, ModuleInfo>> GetModulesByPathForPid(uint32_t pid) {
```
Comment (dimitry-) [inline src/ApiLoader/EnableInTracee.cpp:33]: we probably should keep this int32_t, this is linux specific code and uint32_t could end-up being out of bound.

In general uint32_t is fine in the interface but once passed to linux-specific code it should be translated to int32_t so we do not have CHECKs for overflow everywhere.

File: src/OrbitBase/ExecutablePathLinux.cpp [hunk 4]
```diff
@@ -27,7 +27,7 @@ std::filesystem::path GetExecutablePath() {
   return std::filesystem::path(std::string(buffer, length));
 }
 
-ErrorMessageOr<std::filesystem::path> GetExecutablePath(int32_t pid) {
+ErrorMessageOr<std::filesystem::path> GetExecutablePath(uint32_t pid) {
   char buffer[PATH_MAX];
```
Comment (dimitry-) [inline src/OrbitBase/ExecutablePathLinux.cpp:31]: check that pid is < MAX_INT

File: src/ObjectUtils/LinuxMap.cpp [hunk 5]
```diff
@@ -78,8 +78,8 @@ ErrorMessageOr<ModuleInfo> CreateModule(const std::filesystem::path& module_path
   return module_info;
 }
 
-ErrorMessageOr<std::vector<ModuleInfo>> ReadModules(int32_t pid) {
-  std::filesystem::path proc_maps_path{absl::StrFormat("/proc/%d/maps", pid)};
+ErrorMessageOr<std::vector<ModuleInfo>> ReadModules(uint32_t pid) {
+  std::filesystem::path proc_maps_path{absl::StrFormat("/proc/%u/maps", pid)};
```
Comment (dimitry-) [inline src/ObjectUtils/LinuxMap.cpp:82]: CHECK(pid <= MAX_INT)

File: src/MemoryTracing/MemoryInfoProducer.cpp [hunk 6]
```diff
@@ -63,7 +63,7 @@ void SystemMemoryInfoProducerRunFn(MemoryInfoListener* listener, int32_t /*pid*/
 
 std::unique_ptr<MemoryInfoProducer> CreateSystemMemoryInfoProducer(MemoryInfoListener* listener,
                                                                    uint64_t sampling_period_ns,
-                                                                   int32_t pid) {
+                                                                   uint32_t pid) {
```
Comment (dimitry-) [inline src/MemoryTracing/MemoryInfoProducer.cpp:66]: this also should be int32_t since this is linux-specific component.

Also every time in linux code we use uint32_t as pid we should CHECK that it is <= MAX_INT

File: src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp (hunk 1 above)
Comment (pierricgimmig) [reply src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp:333]: Yes good point, thanks. Done

File: src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp (hunk 2 above)
Comment (pierricgimmig) [reply src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp:343]: Absolutely, thanks. Done.

File: src/MemoryTracing/MemoryInfoProducer.cpp (hunk 6 above)
Comment (pierricgimmig) [reply src/MemoryTracing/MemoryInfoProducer.cpp:66]: Yes I agree, done. Also going through all the similar occurences and reverting to int32_t where possible.

File: src/ObjectUtils/LinuxMap.cpp (hunk 5 above)
Comment (pierricgimmig) [reply src/ObjectUtils/LinuxMap.cpp:82]: reverted to using int32_t

File: src/OrbitBase/ExecutablePathLinux.cpp (hunk 4 above)
Comment (pierricgimmig) [reply src/OrbitBase/ExecutablePathLinux.cpp:31]: reverted to using int32_t

File: src/ApiLoader/EnableInTracee.cpp (hunk 3 above)
Comment (pierricgimmig) [reply src/ApiLoader/EnableInTracee.cpp:33]: I agree, reverted to int32_t.

File: src/OrbitBase/ExecutablePathLinux.cpp (hunk 4 above)
Comment (florian-kuebler) [reply src/OrbitBase/ExecutablePathLinux.cpp:31]: Shouldn't we then use `pid_t` instead of `int32_t`?

File: src/OrbitBase/include/OrbitBase/ThreadUtils.h [hunk 7]
```diff
@@ -16,17 +16,69 @@
 
 namespace orbit_base {
 
-#ifdef _WIN32
+static constexpr uint32_t kInvalidThreadId = 0xffffffff;
```
Comment (florian-kuebler) [inline src/OrbitBase/include/OrbitBase/ThreadUtils.h:19]: Why are those moved from ThreadConstants? While it is fine for me to have them here, it feels a bit strange that now we have two places for thread constants.

File: src/Service/MemoryInfoHandler.cpp [hunk 8]
```diff
@@ -18,19 +19,21 @@ void MemoryInfoHandler::Start(orbit_grpc_protos::CaptureOptions capture_options)
   SetEnableCGroupMemory(true);
   SetEnableProcessMemory(true);
 
+  const int32_t pid = orbit_base::GetNativeProcessId(capture_options.pid());
```
Comment (florian-kuebler) [inline src/Service/MemoryInfoHandler.cpp:22]: Nit: pid_t?

**Review by florian-kuebler — APPROVED** [review]
LGTM, thanks!

File: src/OrbitBase/ThreadUtilsLinux.cpp [hunk 9]
```diff
@@ -22,12 +22,45 @@ namespace orbit_base {
...
+pid_t GetCurrentThreadIdNative() {
   thread_local pid_t current_tid = syscall(__NR_gettid);
   return current_tid;
 }
 
-std::string GetThreadName(pid_t tid) {
+pid_t GetCurrentProcessIdNative() { return getpid(); }
+
+uint32_t GetThreadIdFromNative(pid_t tid) {
+  return tid >= 0 ? static_cast<uint32_t>(tid) : orbit_base::kInvalidThreadId;
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsLinux.cpp:46]: I think invalid input should lead to checkfail...

CHECK(tid == -1 || tid >= 0);

and then you can just static_cast.
same thing for GetNative

File: src/OrbitBase/ThreadUtilsLinux.cpp [hunk 10]
```diff
@@ -22,12 +22,45 @@ namespace orbit_base {
...
+
+uint32_t GetProcessIdFromNative(pid_t pid) {
+  return pid >= 0 ? static_cast<uint32_t>(pid) : orbit_base::kInvalidProcessId;
+}
+
+pid_t GetNativeThreadId(uint32_t tid) {
+  return IsValidThreadId(tid) ? static_cast<pid_t>(tid) : kInvalidLinuxThreadId;
+}
+
+pid_t GetNativeProcessId(uint32_t pid) {
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsLinux.cpp:57]: nit: ToNative?

File: src/OrbitBase/include/OrbitBase/ThreadUtils.h (hunk 7 above)
Comment (pierricgimmig) [reply src/OrbitBase/include/OrbitBase/ThreadUtils.h:19]: Moved back to ThreadConstants.h and include that header in ThreadUtils.h.

File: src/Service/MemoryInfoHandler.cpp (hunk 8 above)
Comment (pierricgimmig) [reply src/Service/MemoryInfoHandler.cpp:22]: yep, done.

File: src/OrbitBase/ExecutablePathLinux.cpp (hunk 4 above)
Comment (pierricgimmig) [reply src/OrbitBase/ExecutablePathLinux.cpp:31]: yep done.

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 9 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsLinux.cpp:46]: Done, thanks

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 10 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsLinux.cpp:57]: Yeah, I changed to "ToNativeX" and "FromNativeX", I think it better conveys the fact that it's a conversion function.

File: src/MemoryTracing/MemoryTracingUtils.cpp [hunk 11]
```diff
@@ -214,12 +214,12 @@ ErrorMessageOr<int64_t> ExtractRssAnonFromProcessStatus(std::string_view status_
   return ErrorMessage("RssAnon value not found in the file content.");
 }
 
-ErrorMessageOr<ProcessMemoryUsage> GetProcessMemoryUsage(int32_t pid) noexcept {
+ErrorMessageOr<ProcessMemoryUsage> GetProcessMemoryUsage(uint32_t pid) noexcept {
```
Comment (dimitry-) [inline src/MemoryTracing/MemoryTracingUtils.cpp:217]: Isn't this linux specific API - in which case it should probably use native pid types.

File: src/MemoryTracing/MemoryTracingUtils.cpp [hunk 12]
```diff
@@ -341,10 +341,10 @@ ErrorMessageOr<void> UpdateCGroupMemoryUsageFromMemoryStat(std::string_view memo
   return outcome::success();
 }
 
-ErrorMessageOr<CGroupMemoryUsage> GetCGroupMemoryUsage(int32_t pid) noexcept {
+ErrorMessageOr<CGroupMemoryUsage> GetCGroupMemoryUsage(uint32_t pid) noexcept {
```
Comment (dimitry-) [inline src/MemoryTracing/MemoryTracingUtils.cpp:344]: same here... int32_t

File: src/OrbitBase/ThreadUtilsLinux.cpp [hunk 13]
```diff
@@ -22,12 +22,47 @@ namespace orbit_base {
...
+  CHECK(tid == kInvalidLinuxThreadId || tid >= 0);
+  return static_cast<uint32_t>(tid);
+}
+
+uint32_t FromNativeProcessId(pid_t pid) {
+  CHECK(pid == kInvalidLinuxProcessId || pid >= 0);
+  return static_cast<uint32_t>(pid);
+}
+
+pid_t ToNativeThreadId(uint32_t tid) {
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsLinux.cpp:55]: same check here and below in ToNativeProcessId?
```
CHECK(tid <= kIntMax || tid == kInvalidThreadId);
```

File: src/OrbitBase/ThreadUtilsTest.cpp [hunk 14]
```diff
@@ -108,3 +98,43 @@ TEST(ThreadUtils, GetThreadName) {
...
+
+  for (auto invalid_native_thread_id : invalid_native_thread_ids) {
+    uint32_t tid = orbit_base::FromNativeThreadId(invalid_native_thread_id);
+    EXPECT_FALSE(orbit_base::IsValidThreadId(tid)) << "tid == " << tid;
+  }
+
+  for (auto invalid_native_process_id : invalid_native_process_ids) {
+    uint32_t pid = orbit_base::FromNativeProcessId(invalid_native_process_id);
+    EXPECT_FALSE(orbit_base::IsValidProcessId(pid)) << "pid == " << pid;
+  }
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsTest.cpp:139]: Add DEATH tests for CHECK fails

File: src/OrbitBase/ThreadUtilsWindows.cpp [hunk 15]
```diff
@@ -12,6 +12,65 @@
...
+// ids rely on it. https://devblogs.microsoft.com/oldnewthing/20080228-00/?p=23283
+inline [[nodiscard]] bool IsMultipleOfFour(uint32_t value) { return (value & 3) == 0; }
+
+[[nodiscard]] bool IsValidThreadIdNative(uint32_t tid) {
+  return tid != kInvalidWindowsThreadId && IsMultipleOfFour(tid);
+}
+
+[[nodiscard]] bool IsValidProcessIdNative(uint32_t pid) {
+  return pid != kInvalidWindowsProcessId_0 && pid != kInvalidWindowsProcessId_1 &&
+         IsMultipleOfFour(pid);
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsWindows.cpp:32]: this should probably be a CHECK

File: src/OrbitBase/ThreadUtilsWindows.cpp [hunk 16]
```diff
@@ -12,6 +12,65 @@
...
+static constexpr uint32_t kInvalidWindowsThreadId = 0;
+static constexpr uint32_t kInvalidWindowsProcessId_0 = 0;
+static constexpr uint32_t kInvalidWindowsProcessId_1 = 0xffffffff;
+
+namespace {
+
+// On Windows, thread and process ids are observed to be multiples of 4. Even though there is no
+// formal guarantee for this property, the current implementation of cross platform thread process
+// ids rely on it. https://devblogs.microsoft.com/oldnewthing/20080228-00/?p=23283
+inline [[nodiscard]] bool IsMultipleOfFour(uint32_t value) { return (value & 3) == 0; }
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsWindows.cpp:24]: constexpr?

File: src/Service/ServiceUtils.cpp [hunk 17]
```diff
@@ -410,7 +410,7 @@ ErrorMessageOr<fs::path> FindSymbolsFilePath(
   return ErrorMessage("Unknown object file type.");
 }
 
-bool ReadProcessMemory(int32_t pid, uintptr_t address, void* buffer, uint64_t size,
+bool ReadProcessMemory(uint32_t pid, uintptr_t address, void* buffer, uint64_t size,
```
Comment (dimitry-) [inline src/Service/ServiceUtils.cpp:413]: this is linux-specific, use pid_t?

**Review by dimitry- — COMMENTED** [review]
We do not expect the code to send windows pids to linux service and vice versa - therefore we should probably have stricter check since if this occurs it is certainly a bug.

File: src/OrbitBase/ThreadUtilsWindows.cpp (hunk 16 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsWindows.cpp:24]: This takes in runtime values, we can't use constexpr right?

File: src/MemoryTracing/MemoryTracingUtils.cpp (hunk 11 above)
Comment (pierricgimmig) [reply src/MemoryTracing/MemoryTracingUtils.cpp:217]: Done, thanks.

File: src/MemoryTracing/MemoryTracingUtils.cpp (hunk 12 above)
Comment (pierricgimmig) [reply src/MemoryTracing/MemoryTracingUtils.cpp:344]: Done.

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 13 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsLinux.cpp:55]: Done

File: src/Service/ServiceUtils.cpp (hunk 17 above)
Comment (pierricgimmig) [reply src/Service/ServiceUtils.cpp:413]: It is but it's called through a GetProcessMemoryRequest which uses uint32_t. Added `CHECK(orbit_base::IsValidProcessId(pid));`

File: src/OrbitBase/ThreadUtilsWindows.cpp (hunk 15 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsWindows.cpp:32]: Done.

File: src/OrbitBase/ThreadUtilsLinux.cpp [hunk 18]
```diff
@@ -53,10 +53,12 @@ uint32_t FromNativeProcessId(pid_t pid) {
 }
 
 pid_t ToNativeThreadId(uint32_t tid) {
+  CHECK(tid <= kIntMax || tid == kInvalidThreadId);
   return IsValidThreadId(tid) ? static_cast<pid_t>(tid) : kInvalidLinuxThreadId;
 }
 
 pid_t ToNativeProcessId(uint32_t pid) {
+  CHECK(pid <= kIntMax || pid == kInvalidProcessId);
   return IsValidProcessId(pid) ? static_cast<pid_t>(pid) : kInvalidLinuxProcessId;
```
Comment (dimitry-) [inline src/OrbitBase/ThreadUtilsLinux.cpp:62]: nit: you can actually drop IsValidProcessId now and just do static_cast, I believe.

File: src/Service/ProcessServiceImpl.cpp [hunk 19]
```diff
@@ -82,8 +83,10 @@ Status ProcessServiceImpl::GetProcessMemory(ServerContext*, const GetProcessMemo
   uint64_t size = std::min(request->size(), kMaxGetProcessMemoryResponseSize);
   response->mutable_memory()->resize(size);
   uint64_t num_bytes_read = 0;
-  if (utils::ReadProcessMemory(request->pid(), request->address(),
-                               response->mutable_memory()->data(), size, &num_bytes_read)) {
+  uint32_t pid = request->pid();
```
Comment (dimitry-) [inline src/Service/ProcessServiceImpl.cpp:86]: Can you just do ToNativeProcessId(request->pid()) here? It already has a check right?

File: src/OrbitBase/ThreadUtilsWindows.cpp (hunk 16 above)
Comment (dimitry-) [reply src/OrbitBase/ThreadUtilsWindows.cpp:24]: I think you still can. It is enough that the function could be evaluated at compile time if the parameter is known.

File: src/Service/ProcessServiceImpl.cpp (hunk 19 above)
Comment (pierricgimmig) [reply src/Service/ProcessServiceImpl.cpp:86]: Well ReadProcessMemory will be implemented for Windows too, we might as well use the platform agnostic pid. I removed the check here and added `ToNativeProcessId` in the implementation of ReadProcessMemory.

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 18 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsLinux.cpp:62]: Yes, same as above, thanks, done.

File: src/OrbitBase/ThreadUtilsTest.cpp (hunk 14 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsTest.cpp:137]: done

File: src/OrbitBase/ThreadUtilsWindows.cpp (hunk 16 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsWindows.cpp]: It's only used in this cpp file, will leave as is, on it's single line :-)

File: src/Service/ProcessServiceImpl.cpp [hunk 20]
```diff
@@ -62,7 +63,7 @@ Status ProcessServiceImpl::GetProcessList(ServerContext*, const GetProcessListRe
 Status ProcessServiceImpl::GetModuleList(ServerContext* /*context*/,
                                          const GetModuleListRequest* request,
                                          GetModuleListResponse* response) {
-  int32_t pid = request->process_id();
+  uint32_t pid = request->process_id();
   LOG("Sending modules for process %d", pid);
 
   const auto module_infos = orbit_object_utils::ReadModules(pid);
```
Comment (dimitry-) [inline src/Service/ProcessServiceImpl.cpp:69]: this call wants a native pid

File: src/Service/ProcessServiceImpl.cpp [hunk 21]
```diff
@@ -62,7 +63,7 @@ Status ProcessServiceImpl::GetProcessList(ServerContext*, const GetProcessListRe
 Status ProcessServiceImpl::GetModuleList(ServerContext* /*context*/,
                                          const GetModuleListRequest* request,
                                          GetModuleListResponse* response) {
-  int32_t pid = request->process_id();
+  uint32_t pid = request->process_id();
```
Comment (dimitry-) [inline src/Service/ProcessServiceImpl.cpp:66]: this should probably be int32_t pid = ToNative...

File: src/Service/ServiceUtils.cpp (hunk 17 above)
Comment (dimitry-) [reply src/Service/ServiceUtils.cpp:414]: Ok, I guess I am most concerned here that we do not have a clear boundary between native and interface level. I would explicitly split it into 2 components even - one linux only another one windows-only and have a proxy which will depending on compile flag call one or the other. This proxy would also be responsible for converting pids from interface type to platform specific one.

This is probably not for this PR, since this one is big as it is.

File: src/UserSpaceInstrumentation/InstrumentProcess.cpp [hunk 22]
```diff
@@ -153,7 +154,7 @@ class InstrumentedProcess {
 ErrorMessageOr<std::unique_ptr<InstrumentedProcess>> InstrumentedProcess::Create(
     const CaptureOptions& capture_options) {
   std::unique_ptr<InstrumentedProcess> process(new InstrumentedProcess());
-  const pid_t pid = capture_options.pid();
+  const pid_t pid = orbit_base::ToNativeProcessId(capture_options.pid());
```
Comment (dimitry-) [inline src/UserSpaceInstrumentation/InstrumentProcess.cpp:157]: I wish we passed pid and a list of instrumented functions here instead of capture_options. In which case the conversion could have been done on the caller side.

File: src/Service/ServiceUtils.cpp (hunk 17 above)
Comment (dimitry-) [reply src/Service/ServiceUtils.cpp:414]: Maybe add a TODO here to this regard so we do not forget.

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: src/OrbitBase/ThreadUtilsWindows.cpp [hunk 23]
```diff
@@ -12,6 +12,55 @@
...
+  return tid != kInvalidWindowsThreadId ? tid : orbit_base::kInvalidThreadId;
+}
+
+uint32_t FromNativeProcessId(uint32_t pid) {
+  bool is_invalid_pid = (pid == kInvalidWindowsProcessId_0 || pid == kInvalidWindowsProcessId_1);
+  CHECK(IsMultipleOfFour(pid) || is_invalid_pid);
+  return !is_invalid_pid ? pid : orbit_base::kInvalidProcessId;
+}
+
+uint32_t ToNativeThreadId(uint32_t tid) {
```
[the remark this answers is not in the document]
Comment (dimitry-) [reply src/OrbitBase/ThreadUtilsWindows.cpp:54]: and a DEATH test for this

File: src/Service/ServiceUtils.cpp (hunk 17 above)
Comment (pierricgimmig) [reply src/Service/ServiceUtils.cpp:414]: Yep agreed. This will come up very soon, with the PR for OrbitService on Windows.

File: src/MemoryTracing/MemoryTracingIntegrationTest.cpp [hunk 24]
```diff
@@ -70,7 +70,7 @@ class MemoryTracingIntegrationTestFixture {
     listener_->SetSamplingPeriodNs(memory_sampling_period_ns_);
     // Collect the cgroup memory information only if the process's memory cgroup and the cgroup
     // memory.stat file can be found successfully
-    int32_t pid = static_cast<int32_t>(orbit_base::GetCurrentProcessId());
+    uint32_t pid = orbit_base::GetCurrentProcessId();
```
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/MemoryTracing/MemoryTracingIntegrationTest.cpp:73]: yes, thanks.

File: src/MemoryTracing/include/MemoryTracing/MemoryInfoProducer.h [hunk 25]
```diff
@@ -20,7 +20,7 @@ using MemoryInfoProducerRunFn = std::function<void(MemoryInfoListener*, int32_t)
 // specified in sampling_period_ns_, and send the extracted information to the listener.
 class MemoryInfoProducer {
  public:
-  explicit MemoryInfoProducer(uint64_t sampling_period_ns, int32_t pid, MemoryInfoProducerRunFn fn)
+  explicit MemoryInfoProducer(uint64_t sampling_period_ns, uint32_t pid, MemoryInfoProducerRunFn fn)
```
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/MemoryTracing/include/MemoryTracing/MemoryInfoProducer.h:23]: thanks.

[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/MemoryTracing/include/MemoryTracing/MemoryInfoProducer.h:23]: done.

File: src/Service/ProcessServiceImpl.cpp (hunk 20 above)
Comment (pierricgimmig) [reply src/Service/ProcessServiceImpl.cpp:69]: yes, thanks, done.

File: src/Service/ProcessServiceImpl.cpp (hunk 21 above)
Comment (pierricgimmig) [reply src/Service/ProcessServiceImpl.cpp:66]: done.

File: src/UserSpaceInstrumentation/InstrumentProcess.cpp (hunk 22 above)
Comment (pierricgimmig) [reply src/UserSpaceInstrumentation/InstrumentProcess.cpp:157]: yeah...

File: src/OrbitBase/ThreadUtilsWindows.cpp (hunk 23 above)
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsWindows.cpp:54]: Added the check, will tidy up the tests in next PR.

File: src/OrbitBase/ThreadUtilsWindows.cpp [hunk 26]
```diff
@@ -12,6 +12,55 @@
...
+  return IsValidThreadId(tid) ? tid : kInvalidWindowsThreadId;
+}
+
+uint32_t ToNativeProcessId(uint32_t pid) {
```
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsWindows.cpp:59]: Done.
